### PR TITLE
Track oldest config generation used for streaming search.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h [hunk 1]
```diff
@@ -16,13 +16,15 @@ class SearchEnvironmentSnapshot
     std::shared_ptr<const RankManager::Snapshot> _rank_manager_snapshot;
     std::shared_ptr<VsmfieldsConfig>             _vsm_fields_cfg;
     std::shared_ptr<const vsm::DocsumTools>      _docsum_tools;
+    int64_t                                      _generation;
```
Comment (geirst) [inline streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h:19]: Consider rename to `_config_generation`.

File: streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h [hunk 2]
```diff
@@ -16,13 +16,15 @@ class SearchEnvironmentSnapshot
...
 
 public:
-    SearchEnvironmentSnapshot(const RankManager& rank_manager, const vsm::VSMAdapter& vsm_adapter);
+    SearchEnvironmentSnapshot(const RankManager& rank_manager, const vsm::VSMAdapter& vsm_adapter, int64_t generation);
     ~SearchEnvironmentSnapshot();
     const std::shared_ptr<const RankManager::Snapshot>& get_rank_manager_snapshot() const noexcept { return _rank_manager_snapshot; }
     const std::shared_ptr<VsmfieldsConfig>& get_vsm_fields_config() const noexcept { return _vsm_fields_cfg; }
     const std::shared_ptr<const vsm::DocsumTools>& get_docsum_tools() const noexcept { return _docsum_tools; }
+    int64_t get_generation() const noexcept { return _generation; }
```
Comment (geirst) [inline streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h:27]: Consider rename to `get_config_generation()`.

**Review by geirst — DISMISSED** [review]
👍

File: streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h (hunk 1 above)
Comment (toregge) [reply streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h]: Renamed.

File: streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h (hunk 2 above)
Comment (toregge) [reply streamingvisitors/src/vespa/searchvisitor/search_environment_snapshot.h]: Renamed.

**Review by geirst — APPROVED** [review]
👍

**Comment by geirst** [comment]
@arnej27959 FYI

**Pull request merged**